libflux/src/core/parser/tests.rs

```diff
@@ -48,6 +48,534 @@ impl<'a> Locator<'a> {
+                        location: loc.get(1, 18, 1, 19),
+                        ..BaseNode::default()
+                    },
+                    expression: Expression::Identifier(Identifier {
+                        base: BaseNode {
+                            location: loc.get(1, 18, 1, 19),
+                            ..BaseNode::default()
+                        },
+                        name: "g".to_string(),
+                    })
+                })),
+            ],
+            eof: None,
+        },
+    )
+}
+
+#[test]
+fn parse_greedy_quotes_paren_wrapped() {
+    let mut p = Parser::new(r#"(“some string”)"#);
+    let parsed = p.parse_file("".to_string());
+    let loc = Locator::new(&p.source[..]);
+    assert_eq!(
+        parsed,
+        File {
+            base: BaseNode {
+                location: loc.get(1, 1, 1, 20),
+                ..BaseNode::default()
+            },
+            name: "".to_string(),
+            metadata: "parser-type=rust".to_string(),
+            package: None,
+            imports: vec![],
+            body: vec![Statement::Expr(Box::new(ExprStmt {
+                base: BaseNode {
+                    location: loc.get(1, 1, 1, 20),
+                    ..BaseNode::default()
+                },
+                expression: Expression::Paren(Box::new(ParenExpr {
+                    base: BaseNode {
+                        location: loc.get(1, 1, 1, 20),
+                        errors: vec!["invalid expression @1:16-1:19: ”".to_string()],
+                        ..BaseNode::default()
+                    },
+                    lparen: None,
+                    expression: Expression::Binary(Box::new(BinaryExpr {
+                        base: BaseNode {
+                            location: loc.get(1, 5, 1, 16),
+                            ..BaseNode::default()
+                        },
+                        operator: Operator::InvalidOperator,
+                        left: Expression::Identifier(Identifier {
+                            base: BaseNode {
+                                location: loc.get(1, 5, 1, 9),
+                                errors: vec!["invalid expression @1:2-1:5: “".to_string()],
+                                ..BaseNode::default()
+                            },
+                            name: "some".to_string(),
+                        }),
+                        right: Expression::Identifier(Identifier {
+                            base: BaseNode {
+                                location: loc.get(1, 10, 1, 16),
+                                ..BaseNode::default()
+                            },
+                            name: "string".to_string(),
+                        })
+                    })),
+                    rparen: None,
+                }))
+            }))],
+            eof: None,
+        },
+    )
+}
+
+#[test]
+fn parse_greedy_quotes_bare() {
+    let mut p = Parser::new(r#"“some string”"#);
```

Comment:
This one brings up an interesting case. Should we allow any valid utf8 sequence of characters in a string literal?

According to the SPEC we state that strings can have any character https://github.com/influxdata/flux/blob/master/docs/SPEC.md#string-literals 

Thoughts? Should we create a new issue to fix this?

Reply:
I think it would be tricky to do that without changing the Ragel file. I think this test string is slightly different than what you describe though. In this case, the invalid quotes are wrapping the user-defined string, but are not a part of it. Unless I'm misunderstanding?

I'm not opposed to having any character defined within a string though. Could be useful.

Reply:
Lets create an issue around this and then we can groom and see what it would take to do.

Reply:
Issue created: https://github.com/influxdata/flux/issues/3206